plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/3dtiles/BuildingFilter/hooks.ts

```diff
@@ -0,0 +1,52 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { BaseFieldProps } from "../../types";
+
+import { useBuildingFilter } from "./useBuildingFilter";
+
+type OptionsState = Omit<BaseFieldProps<"buildingFilter">["value"], "id" | "group" | "type">;
+
+const useHooks = ({
+  value,
+  dataID,
+  onUpdate,
+}: Pick<BaseFieldProps<"buildingFilter">, "value" | "dataID" | "onUpdate">) => {
+  const [options, setOptions] = useState<OptionsState>({
+    height: value.height,
+    abovegroundFloor: value.abovegroundFloor,
+    basementFloor: value.basementFloor,
+  });
+
+  const handleUpdate = useCallback(
+    <P extends keyof OptionsState>(prop: P, v?: OptionsState[P]) => {
+      setOptions(o => {
+        const next = { ...o, [prop]: v };
+        onUpdate({ id: value.id, type: value.type, group: value.group, ...next });
+        return next;
+      });
+    },
+    [onUpdate, value],
+  );
+
+  const handleUpdateRange = useCallback(
+    (prop: keyof OptionsState) => (value: number | number[]) => {
+      if (value && Array.isArray(value)) {
+        handleUpdate(prop, value as [from: number, to: number]);
+      }
+    },
+    [handleUpdate],
+  );
+
+  useEffect(() => {
+    setOptions({ ...value });
+  }, [value, onUpdate]);
```

Comment:
I don't think you want `options` to be set to value, right?
  `options` type is value's type minus a few fields right? 
Also, won't this useEffect make any updates call setOptions twice (once in the handleUpdate function and then again here in the useEffect when the value change triggers it)??

Reply:
I fixed.
[fix: prevent unnecessary render](https://github.com/eukarya-inc/reearth-plateauview/pull/121/commits/44fd8f17ea04a3832dedcc71dae046af0b73de61)